model/NewSqlResultStorage.php

```diff
@@ -57,6 +58,7 @@ protected function prepareVariableDataForSchema($deliveryResultIdentifier, $test
             self::VARIABLE_VALUE => $serializedVariable,
             self::VARIABLE_HASH => $deliveryResultIdentifier . md5($deliveryResultIdentifier . $serializedVariable . $callId),
             self::CREATED_AT => $createdAt->format($persistence->getPlatform()->getDateTimeFormatString()),
+            self::IS_EXTERNALLY_GRADED => (int)$variable->getExternallyGraded(),
```

Comment:
Why are we introducing a new column to manage this instead of just adding a new property in the JSON we save on `value` column? If we save it there, might be easier to make it compatible with other drivers in the future. 

Today we already managed everything on that JSON inside the value column. 

Also keep in mind that if we need other "flags" like this in the future, it becomes a bit hard to maintain keeping adding columns in the DB.

Reply:
are you sure of that @gabrielfs7 ??? because I am not sure of that.

Reply:
If you tell me for sure that we can extend it, then yes. I can change it.

Reply:
I did a test locally and wrote in the `variables_storage.value` for `identifier=SCORE` the following:

```
{"isExternallyGraded":true,"identifier":"SCORE","cardinality":"single","baseType":"float","epoch":"0.01757600 1678811051","type":"outcomeVariable","normalMinimum":null,"normalMaximum":null,"value":"MA=="}
```

I was still able to view and export the result in the UI, so I think it does not break the implementation.

Reply:
Do we have other places where we use it? Do you know who knows it?

Reply:
As far as I know, we jsondecode/unserialize to be user later by classes like I shared with you (i.e `taoResultServer_models_classes_Variable`).

https://github.com/oat-sa/extension-tao-outcomerds/blob/master/model/AbstractRdsResultStorage.php#L647 (here where we get the values from persistence)

https://github.com/oat-sa/extension-tao-outcome/blob/master/models/classes/class.Variable.php#L239 (getting data from the json and factoring the response variables) 

https://github.com/oat-sa/extension-tao-outcome/blob/master/models/classes/class.Variable.php#L182 (and here it is how it persists in the DB).

IMO seems that regardless the json values existing there, it only consider the ones mapped in the PHP classes. So, it seems if you add new valid json attributes, it still works :)

Reply:
Thanks, @gabrielfs7 . Let me take a look.